### ISSUE TITLE
Flex doesn't correctly parse Reference Object in Parameters section. It's not parsing the referenced Parameter object?

### ISSUE THREAD

**Issue by EBichel** [pipermerriam/flex#80]
Hi,
I've found that flex doesn't correctly validate a referenced parameter object.
An example to reproduce the error:
```json
{
  "swagger" : "2.0",
  "info" : {
    "version" : "0.1",
    "title" : "SampleAPI"
  },
  "basePath" : "/rest",
  "schemes" : [ "http" ],
  "consumes" : [ "application/json" ],
  "produces" : [ "application/json" ],
  "paths" : {
    "/resource": {
      "post" : {
        "parameters" : [
            {"$ref" : "#/parameters/ParameterObject"}
        ],
        "responses" : {
          "200" : {
            "description" : "ok"
          }
        }
      }
    }

  },
  "parameters" : {
    "ParameterObject" : {
      "name" : "Parameter Object",
      "description" : "Fancy description",
      "in" : "body",
      "required" : true,
      "schema" : {
        "type" : "object",
        "properties" : {
          "ArrayOfItems" : {
            "type" : "array",
            "items" : {
              "$ref" : "#/definitions/RandomObject"
            }
          }
        }
      }
    }
      
  },
  "definitions" : {

    "RandomObject" : {
      "type" : "object",
      "properties" : {
        "id" : {
          "type" : "string"
        },
        "name" : {
          "type" : "string"
        },
        "version" : {
          "type" : "string"
        }
      }
    }
  }
}
```

I get the following output from flex: 
```
Error: 'paths':
    - u'/resource':
        - 'post':
            - 'parameters':
                - 'required':
                    - 'name':
                        - u'This value is required'
                    - 'in':
                        - u'This value is required'
```

Obviously `name` and `in` are existing. I think this is easy to fix, since it seems to me that the reference object is validated instead of the parameter object that is referenced.
Thanks a lot for the good work and fast feedback!

**Comment by pipermerriam** [comment]
What version of flex are you using?  I'm having trouble reproducing this.

**Comment by EBichel** [comment]
I am sorry. I was using version 4.0.1.. This is fixed in 4.1
